File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.swift
import Foundation

func solution(_ players:[String], _ callings:[String]) -> [String] {
    var result: [String] = players
    var dict: [String : Int] = [:]
    for (index, name) in players.enumerated() {
        dict[name] = index
    }
    for i in 0..<callings.count {
        let calledPlayer = callings[i] 
        let prePlayer = result[dict[calledPlayer]! - 1] 
        dict.updateValue(dict[calledPlayer]! - 1, forKey: calledPlayer)
        dict.updateValue(dict[calledPlayer]! + 1, forKey: prePlayer)
        result[dict[calledPlayer]!] = calledPlayer
        result[dict[prePlayer]!] = prePlayer
    }
    
    return result
}

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    solDict = {}
    answer = players
    for index, name in enumerate(players):
        solDict[name] = index;
    for i in range(len(callings)):
        called_player = callings[i] # "kai"
        pre_called_player = answer[solDict.get(called_player) - 1] # "poe"
        solDict[called_player] = solDict.get(called_player) - 1 
        solDict[pre_called_player] = solDict.get(pre_called_player) + 1;
        answer[solDict.get(called_player)] = called_player
        answer[solDict.get(pre_called_player)] = pre_called_player
    return answer